python/pyarrow/_compute.pyx

```diff
@@ -2601,7 +2612,11 @@ def register_scalar_function(func, function_name, function_doc, in_types,
 
     c_func_name = tobytes(function_name)
 
-    func_spec = inspect.getfullargspec(func)
+    try:
+        func_spec = inspect.getfullargspec(func)
+        is_varargs = func_spec.varargs is not None
+    except:
```

Comment:
`except` what? Bare `except` statements are discouraged, you should catch the precise exception type here.
Also, please add a comment explaining when this can happen.